### PR TITLE
handle case when exclude_dictionary is empty

### PR DESCRIPTION
**Pull request by dsego**

### CODE REVIEW

**Comment by dsego** [comment]
@dino-cikatic @mattdrayer Can you review, please?

File: search/tests/tests.py [hunk 1]
```diff
@@ -744,6 +744,23 @@ def test_exclude_filter_multiple(self):
...
 
+    def test_exclude_filter_empty(self):
+        """ Test that search works when exclude filter is an empty list """
+        self.searcher.index("test_doc", {"course": "ABC", "org": "edX", "id": "FAKE_ID_1"})
+        self.searcher.index("test_doc", {"course": "XYZ", "org": "edX", "id": "FAKE_ID_2"})
+        self.searcher.index("test_doc", {"course": "DEF", "org": "MITX", "id": "FAKE_ID_3"})
+        self.searcher.index("test_doc", {"course": "GHI", "org": "HarvardX", "id": "FAKE_ID_4"})
+        self.searcher.index("test_doc", {"course": "LMN", "org": "edX", "id": "FAKE_ID_5"})
+
+        response = self.searcher.search(exclude_dictionary={"org": []})
```
Comment (dino-cikatic) [inline search/tests/tests.py]: Can we also add a test that would run query with filter/field dictionary and empty exclude dictionary? Just to be sure..

**Comment by dino-cikatic** [comment]
Additional asserts look good. :+1:  from me.

File: search/elastic.py [hunk 2]
```diff
@@ -135,6 +135,9 @@ def _process_exclude_dictionary(exclude_dictionary):
             exclude_values = [exclude_values]
         not_properties.extend([{"term": {exclude_property: exclude_value}} for exclude_value in exclude_values])
 
+    if not not_properties:
```
Comment (mattdrayer) [inline search/elastic.py]: Can you add some more information/context in the docstring and or comments around what 'not_properties' represents, how it's used, expected scenarios, what it means to return empty set, etc.

**Comment by mattdrayer** [comment]
LGTM -- I'd just like to see some more info/context added to this method, because it's not exactly clear to me when not_properites is valid and when the negative case is encountered.
